server/chat-commands/info.ts

```diff
@@ -2429,52 +2432,101 @@ export const commands: ChatCommands = {
-			let width = targets[1].trim();
-			if (!width) return this.errorReply(`No width for the image was provided!`);
-			if (!isNaN(parseInt(width))) width += `px`;
-
-			let height = targets[2].trim();
-			if (!height) return this.errorReply(`No height for the image was provided!`);
-			if (!isNaN(parseInt(height))) height += `px`;
-
-			const unitRegex = /^\d+(?:p[xtc]|%|[ecm]m|ex|in)$/;
-			if (!unitRegex.test(width)) {
-				return this.errorReply(`"${width}" is not a valid width value!`);
-			}
-			if (!unitRegex.test(height)) {
-				return this.errorReply(`"${height}" is not a valid height value!`);
-			}
+		return this.errorReply(`/showimage has been deprecated - use /show instead.`);
+	},
 
-			return this.sendReply(Utils.html`|raw|<img src="${image}" style="width: ${width}; height: ${height}" />`);
+	requestshow(target, room, user) {
+		if (!this.canTalk()) return false;
+		if (!room.settings.approvalsEnabled) {
+			return this.errorReply(`Media approvals are disabled in this room.`);
+		}
+		if (this.can('mute', null, room)) return this.errorReply(`Use !show instead.`);
+		if (room.pendingApprovals?.has(user.id)) return this.errorReply('You have a request pending already.');
+		if (!toID(target)) return this.parse(`/help requestshow`);
+		if (!/^https?:\/\//.test(target)) target = `https://${Utils.escapeHTML(target)}`;
+		if (!room.pendingApprovals) room.pendingApprovals = new Map();
+		room.pendingApprovals.set(user.id, target);
+		this.sendReply(`You have requested to show the link: ${target}`);
+		room.sendMods(
+			`|uhtml|request-${user.id}|<div class="infobox">${user.name} has requested approval to show <a href="${target}">${target}</a><br>` +
+			`<button class="button" name="send" value="/approveshow ${user.id}">Approve</button><br>` +
+			`<button class="button" name="send" value="/denyshow ${user.id}">Deny</button></div>`
+		);
+		return room.update();
+	},
+	requestshowhelp: [`/requestshow [link], [comment] - Requests permission to show media in the room.`],
+
+	async approveshow(target, room, user) {
+		if (!this.can('mute', null, room)) return false;
+		if (!room.settings.approvalsEnabled) {
+			return this.errorReply(`Media approvals are disabled in this room.`);
+		}
+		target = toID(target);
+		if (!target) return this.parse(`/help approvelink`);
+		const id = room.pendingApprovals?.get(target);
+		if (!id) return this.errorReply(`${target} has no pending request.`);
+		this.privateModAction(`(${user.name} approved ${target}'s media display request.)`);
+		this.modlog(`APPROVESHOW`, null, `${target} (${id})`);
+		const YouTube = new YoutubeInterface();
+		room.pendingApprovals!.delete(target);
+		if (/^(https?:\/\/)?(www\.)?(youtube\.com|youtu\.be)(\/|$)/i.test(id)) {
+			let res = await YouTube.generateVideoDisplay(id);
+			room.sendMods(`|uhtmlchange|request-${target}|`);
+			res += `<br><p style="margin-left: 5px; font-size:9pt;color:white;">(Suggested by ${target})</p>`;
+			this.addBox(res as string);
```

Comment:
Consider that needing to cast `res` is a sign that maybe you forgot to do something............

Reply:
(Specifically, to handle the case where the link is invalid.)